packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts

```diff
@@ -144,3 +145,8 @@ export function calculatePinnedRowsHeight(apiRef: React.MutableRefObject<GridApi
     bottom: bottomPinnedRowsHeight,
   };
 }
+
+export function getMinimalContentHeight(apiRef: React.MutableRefObject<GridApiCommunity>) {
```

Comment:
Wouldn't make more sense to have it in `useGridVirtualScroller`? Then `GridOverlays` could receive the height already calculated as prop.